File: client/Assets/Plugins/Android/java/com/shiyuegame/httprequest/WebRequestInfo.java
package com.shiyuegame.httprequest;

public class WebRequestInfo {

    public String path = null;
    public String outputPath = null;
//    public Action<byte[], string> webSucceCallback = null;
//    public Action<VersionAsset, string> webFailCallback = null;
    public String alertTxt = "";
    public int tryCount = 3;
    public int timeout = 10;
    public boolean useRandom = false;
//    public UnityWebRequest webRequest = null;
//    public UnityWebRequest headRequest = null;
}

File: client/Assets/Plugins/iOS/SySwiftNetWork/WebRequestInfo.swift
//
//  WebRequestInfo.swift
//  TestSwift
//
//  Created by huangyuqiu on 2023/3/16.
//

import Foundation

class WebRequestInfo {
    public var path:String?
    public var outputPath:String?
    public var alertTxt:String?
    public var tryCount:Int?
    public var timeout:Int?
    public var useRandom:Bool = false
}

enum DownloadError: Error {
    case ParseError;
    case DownloadError;
    case DownloadPathIsNilError;
    case DownloadMakeDirError;
    case DownloadResponseStatucCodeError;
}
